packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.types.ts

```diff
@@ -29,12 +29,12 @@ export interface IExpandingCardProps extends IBaseCardProps<IExpandingCard, IExp
   /**
    *  Render function to populate compact content area
    */
-  onRenderCompactCard?: IRenderFunction<IExpandingCardProps>;
```

Comment:
I actually think after looking at how these renderers are used they should not even be of type `IRenderFunction`:
https://github.com/OfficeDev/office-ui-fabric-react/blob/d5320e5be4ddc4ea55ac1b4608cdd0f6df935c33/packages/utilities/src/IRenderFunction.ts#L6-L8
`IRenderFunction` expects some component props to be passed and an optional default renderer... and we are not using them for any of that. The correct type should be: `onRenderCompactCard?: (data: any) => JSX.Element` where the `any` can be further improved as @lijunle stated with some generics.

It could be a breaking change and doable only with a major release possibly. In this case we should put a TODO comment.

Reply:
@Vitalius1 I think it *is* a render function. The only question in my mind is, what is the default renderer in `onRenderCompactCard` case? If it is rendering nothing, so, the default render is `() => false`.

Reply:
I think passing a default renderer as `() = > null` would be an overkill for trying to keep the `IRenderFunction` type on this renderer. Especially if it passes back just one prop `renderData` and not all the props as the `IRenderFunction` is designed to do. Anyways I don't have a strong opinion on what type it should be as long as type safety is nor regressed.

P.S. Could you please include in this PR the same change for the `PlainCard` which is following the same format as the `ExpandingCard`. Thanks!

Reply:
I will update the `PlainCard` typing.

After read the typing for `IRenderFunction`. I think it is a little bit strange. It declares the `props` and `defaultRenderer` as optional. From usage aspect, it needs to check the `defaultRenderer` exists before leverage it, so it is not really *default*. IMO, *default* means it is always available.

I suppose the correct typing should be:

```ts
export interface IRenderFunction<P> {
  (props: P, defaultRender: (props?: P) => JSX.Element | null): JSX.Element | null;
}
```

It should be declare `?` in the callback typing. Fabric-react is passing the value to the callback. Fabric-react knows it is there, and not optional.

/cc @dzearing